File: front/api/DTO/customLayers.dto.ts
/**
 * Représente une couche custom configurée par l'utilisateur.
 */
export interface CustomLayerDTO {
  type: string;
  params: Record<string, number>;
}

/**
 * Liste ordonnée de couches custom pour l'entraînement from scratch.
 */
export interface CustomLayersDTO {
  layers: CustomLayerDTO[];
}

File: back/app/model/domain/DTO/customLayersDTO.py
from pydantic import BaseModel
from typing import Any


class CustomLayerDTO(BaseModel):
    """Représente une couche custom ajoutée par l'utilisateur."""
    type: str
    params: dict[str, Any] = {}


class CustomLayersDTO(BaseModel):
    """Liste ordonnée de couches pour l'entraînement from scratch custom."""
    layers: list[CustomLayerDTO] = []
